x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/DefaultCursor.java

```diff
@@ -67,4 +77,13 @@ public void close() throws SQLException {
             client.queryClose(cursor);
         }
     }
+
+    public List<String> warnings() {
+        return warnings;
+    }
+
+    @Override
+    public Cursor withoutWarnings() {
+        return new DefaultCursor(client, cursor, columnInfos, rows, meta, Collections.emptyList());
```

Comment:
`clearWarnings` method is easy to map the actual functionality and can simply clear the list with a check to be emptyList-friendly:
```
if (list.isEmpty() == false) { 
   list.clear()
}
```

Reply:
`List#clear()` is not implemented in the list instance used here (it's unmodifiable).